src/ko/toon11/src/eu/kanade/tachiyomi/extension/ko/toon11/Toon11.kt

```diff
@@ -0,0 +1,263 @@
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        title = element.selectFirst(".homelist-title")!!.text()
+        element.selectFirst(".homelist-thumb")?.also {
+            thumbnail_url = "https:" + it.attr("style").substringAfter("url('").substringBefore("')")
+        }
+    }
+
+    override fun popularMangaNextPageSelector() = ".pg_end"
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        title = element.selectFirst(".homelist-title")!!.text()
+        val dataId = element.attr("data-id")
+        setUrlWithoutDomain("$baseUrl/bbs/board.php?bo_table=toons&stx=$title&is=$dataId")
+        element.selectFirst(".homelist-thumb")?.also {
+            thumbnail_url = "https:" + it.attr("style").substringAfter("url('").substringBefore("')")
+        }
+    }
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.selectFirst("h2.title")!!.text()
+            thumbnail_url = document.selectFirst("img.banner")?.absUrl("src")
+            document.selectFirst("span:contains(분류) + span")?.also { status = parseStatus(it.text()) }
+            document.selectFirst("span:contains(작가) + span")?.also { author = it.text() }
+            document.selectFirst("span:contains(소개) + span")?.also { description = it.text() }
+            document.selectFirst("span:contains(장르) + span")?.also { genre = it.text().split(",").joinToString() }
+        }
+    }
+
+    private fun parseStatus(element: String): Int = when {
+        "완결" in element -> SManga.COMPLETED
+        "주간" in element || "월간" in element || "연재" in element || "격주" in element -> SManga.ONGOING
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val nav = document.selectFirst("span.pg")
+        val chapters = ArrayList<SChapter>()
+
+        document.select(chapterListSelector()).forEach {
+            chapters.add(chapterFromElement(it))
+        }
+
+        if (nav == null) {
+            return chapters
+        }
+
+        val pg2url = nav.selectFirst(".pg_current ~ .pg_page")!!.absUrl("href")
+
+        // recursively build the chapter list
+
+        fun parseChapters(nextURL: String) {
+            val newpage = fetchPagesFromNav(nextURL)
+            newpage.select(chapterListSelector()).forEach {
+                chapters.add(chapterFromElement(it))
+            }
+            val newURL = newpage.selectFirst(".pg_current ~ .pg_page")?.absUrl("href")
+            if (!newURL.isNullOrBlank()) parseChapters(newURL)
+        }
```

Comment:
put it outside the function body and try adding `tailrec` to optimize the recursion on compile time
`private tailrec fun ...`